reference.md

```diff
@@ -404,31 +403,90 @@ two distinct concepts with similar names.
-its first element and 5 as its third element; if it matches, then it
-evaluates `(print :q q)` with `q` bound to the second element of
-`mytable`. The final clause will only match if `mytable` has 1 as its
-first element; if so then it will add up the second and third elements.
+pattern/body clauses.
+
+The first clause will only match if `mytable` is 59.
+
+The second clause will match if `mytable` is a table with 9 as its first
+element, any non-nil value as its second value and 5 as its third element; if
+it matches, then it evaluates `(print :q q)` with `q` bound to the second
+element of `mytable`.
+
+The final clause will only match if `mytable` has 1 as its first element and
+two non-nil values after it; if so then it will add up the second and third
+elements.
+
+If no clause matches, the form evaluates to nil.
 
-Patterns can be tables, literal values, or symbols.
+Patterns can be tables, literal values, or symbols. Any symbol is implicitly
+checked to be not `nil`. Symbols can be repeated in an expression to check for
+the same value.
 
-Tables can be nested, and they may be either sequential (`[]` style)
-or key/value (`{}` style) tables. Sequential tables will match if they
-have at least as many elements as the pattern. (To allow an element to
-be nil, use a symbol like `?this`.) Tables will never fail to match
-due to having too many elements. You can use `&` to  capture all the
-remaining elements of a sequential table, just like `let`.
+Example:
 
 ```fennel
-(match mytable
+(case mytable
+  ;; the first and second values of mytable are not nil and are the same value
+  [a a] (* a 2)
+  ;; the first and second values are not nil and are not the same value
+  [a b] (+ a b))
+```
+
+It's important to note that expressions are checked *in order!* In the above
+example, since `[a a]` is checked first, we can be confident that when `[a b]`
+is checked, the two values must be different. Had the order been reversed,
+`[a b]` would always match as long as they're not `nil` - even if they have the
+same value!
+
+You may allow a symbol to optionally be `nil` by prefixing it with `?`.
+
+Example:
+
+```fennel
+(case mytable
+  ;; not-nil, maybe-nil
+  [a ?b] :maybe-one-maybe-two-values
+  ;; maybe-nil == maybe-nil
+  [?a ?a] :maybe-none-maybe-two-same-values
+  ;; maybe-nil, maybe-nil
+  [?a ?b] :maybe-none-maybe-one-maybe-two-values)
+```
+
+Symbols may also be prefixed with `_`, which is similar to `?` except repeated
+symbols values are not compared. `_` is also often used on its own at the end
+of an expression to define an "else" style fall-through.
+
+Example:
+
+```fennel
+(case mytable
+  ;; not-nil, maybe-nil
+  [a _b] :maybe-one-maybe-two-values
+  ;; maybe-nil, maybe-nil (different to the last example!)
+  [_a _a] :maybe-none-maybe-one-maybe-two-values
+  ;; maybe-nil, maybe-nil, note this is effectively the same as []
+  [_a _b] :maybe-none-maybe-one-maybe-two-values
```

Comment:
Should clarify here that this clause cannot ever match, because anything that would match here would also match the previous one.

Reply:
https://github.com/bakpakin/Fennel/pull/434/commits/b4e9c5898d23c9671680025b9be1a4c7cc0e7890 

I have left the `_a _b` there as parity with the `?a ?b` example but noted its non-functional.